cs3api4lab/api/cs3_file_api.py

```diff
@@ -129,14 +128,15 @@ def write_file(self, file_path, content, endpoint=None):
 
         time_start = time.time()
 
-        if self.storage_logic.stat(file_path, endpoint):
+        stat = self.storage_api.stat(file_path, endpoint)
+        if stat.status.code == cs3code.CODE_OK:
```

Comment:
Here it makes sense to write even is stat fails, but only for file not existing reasons. If the API fails, maybe we should fail as well.

Reply:
Right now Reva sometimes returns code internal for non-existing files

Reply:
I was even going to ask why we even stat here and not just try to lock. But I guess that when we switch to the lock api, we need the file to exist, since the lock is written in its metadata.

If reva is not consistent, I guess there's no other alternative and I will resolve this discussion, but please add a FIXME here and explain the situation.. Cause we can be overwriting a file just because some network request failed.

Reply:
added fixme in both places